### PR TITLE
Resolve issue #158

### PR DESCRIPTION
**Pull request by ulanzetz**
Add config setting `enableOptCollectionFlattening` to allow transformation from Option[Collection[A]] to Collection[B] if transformation from A to B is defined

### CODE REVIEW

File: chimney/src/main/scala/io/scalaland/chimney/dsl/ConfigDsl.scala [hunk 1]
```diff
@@ -14,6 +14,16 @@ trait ConfigDsl[CC[_ <: TransformerCfg], C <: TransformerCfg] {
...
     this.asInstanceOf[CC[EnableMethodAccessors[C]]]
 
+  /** Enable transformation from Option[Collection[A]] to Collection[B] if transformation from A to B can be derived.
+    * If Option is empty, it will be empty collection
+    *
+    * By default in such case compilation fails.
+    *
+    * @see [[https://scalalandio.github.io/chimney/transformers/customizing-transformers.html#using-method-accessors]] for more details
+    */
+  def enableOptCollectionFlatting: CC[EnableOptCollectionFlatting[C]] =
```
Comment (paulpdaniels) [inline chimney/src/main/scala/io/scalaland/chimney/dsl/ConfigDsl.scala:24]: Nit: Shouldn't it be `Flattening`?

Comment (ulanzetz) [reply chimney/src/main/scala/io/scalaland/chimney/dsl/ConfigDsl.scala]: fixed

**Comment by MateuszKubuszok** [comment]
Thanks for contribution!

To be honest, I would prefer to investigate why `implicit def transformFlatten[A]: Transform[F[G[A], H[A]]` is being ignored. Adding special case in config is a last resort solution, as someone else might do something similar for other collection and have the same issue as #158. So I would prefer to figure out a root cause before we implement special-case workaround.

**Comment by ulanzetz** [comment]
@MateuszKubuszok 
It's being ignored due type parameters. If we define something like this:
```implicit def dependent[A, B](implicit underlying: Transformer[A, B]): Tranformer[...]```,
it will be not derived, if Transformer[A, B] is not defined explicitly (not derived in macro).
Or we should explicitly define `dependent` for each (A, B) pair

```
implicit val dependentStringString: Transformer[...] = dependent[String, String]
implicit val dependentIntInt: Transformer[...] = dependent[Int, Int]
```

For same reason we can't define unsafe option unwrapper without marco
```
implicit def optUnwrapped[A, B](implicit underlying: Transformer[A, B]): Transformer[Option[A], B] =
    a => underlying(a.get)
```
This will also be ignored.

The only thing we can do, to add new branch in macro derviation

**Pull request closed without merging**